spec/traits.dd

```diff
@@ -1053,16 +1053,23 @@ void main()
         types (basic, user-defined, template) are supported.)
         $(LI The lambda function body must contain a single expression
         (no return statement) which contains only numeric values,
-        manifest constants, enum values and function arguments. If the
-        expression contains local variables, function calls or return
-        statements, the function is considered uncomparable.)
+        manifest constants, enum values, function arguments and function
+        calls. If the expression contains local variables or return
+        statements, the function is considered incomparable.)
         )
 
         $(P If these constraints aren't fulfilled, the function is considered
         incomparable and `isSame` returns $(D false).)
 
 $(SPEC_RUNNABLE_EXAMPLE_COMPILE
 ---
+int f() { return 2; }
+void test(alias pred)()
+{
+    // f() from main is a different function from top-level f()
+    static assert(!__traits(isSame, a => a + f(), pred));
```

Comment:
@RazvanN7 this fails at the moment:

```
__stdin.d(6): Error: static assert:  `!true` is false
__stdin.d(23):        instantiated from here: `test!((a) => a + f())`
```

Reply:
Yes, this has unveiled a bug which occurs on template lambdas. I temporarily made the template lambda into a non-template lambda so that this can be merged. I will make a PR tomorrow to fix the bug in the lambdacomp module.